src/OnlineRetailPortal.Services/Services/ProductService.cs

```diff
@@ -1,12 +1,25 @@
-﻿using OnlineRetailPortal.Contracts.Models;
+
+﻿using OnlineRetailPortal.Contracts;
+using System.Collections.Generic;
 using System;
 using System.Collections.Generic;
 using System.Text;
 using System.Threading.Tasks;
 
-namespace OnlineRetailPortal.Services.Services
+namespace OnlineRetailPortal.Services
 {
-    class ProductService 
+   public class ProductService : IProductService
     {
+        public async Task<GetProductsServiceResponse> GetProductsAsync(GetProductsServiceRequest request)
+        {
+            List<Core.Product> product = Core.Product.GetProducts(request.PageNo,request.PageSize);
```

Comment:
This call should be awaited. Any calls to DBs / external services should be an aysnc call.